rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -196,24 +191,27 @@ trait CachedFindRuleNodeStatusReports extends ReportingService with CachedReposi
          * actually don't have rulenodestatutreport. So.
          * We add them in NodeId -> Set()
          */
-        val updated = newStatus.groupBy { _.nodeId }
-        val invalidVersion = expired -- updated.keySet
+        val invalidVersion = expired -- newStatus.keySet
         logger.debug(s"Compliance cache miss (invalid version):[${invalidVersion.map(_.value).mkString(" , ")}],"+
-                               s" cache miss (updated):[${updated.keySet.map(_.value).mkString(" , ")}], "+
+                               s" cache miss (updated):[${newStatus.keySet.map(_.value).mkString(" , ")}], "+
                                s" hit:[${upToDate.map(_.value).mkString(" , ")}]")
-        cache = cache ++ invalidVersion.map(id => (id,Set[RuleNodeStatusReport]())).toMap ++ updated
+        cache = cache ++ invalidVersion.map(id => (id,(NoRunNoInit, Set[RuleNodeStatusReport]()))).toMap ++ newStatus
```

Comment:
OK, so you were right: the code and the comments were inconsistent. At some point, we added garanties that each nodeIds given in inputs get an output (be it NoRunNoVersion). So part of the code is dead code.